### PR TITLE
Add an http mode for publishing and notifying firmware updates

### PR DESCRIPTION
**Pull request by aabadie**
This contains the work done to adapt the current SUIT support of RIOT to make it work with a remote HTTP server managing the firmwares and manifest.

The HTTP server used is available on GitHub here: https://github.com/aabadie/ota-server

A new variable `SUIT_MODE` is added by this PR, it can be either 'local', e.g. when using ethos, or 'http' when using the HTTP ota-server application.
The 'local' mode is the default.

I tried to avoid as much as possible code duplication by factorizing common parts.

The current code is adapted to the signed suit_v4 manifest and needs a specific branch of the otaserver to work, see https://github.com/aabadie/ota-server/pull/5

## How to use

1. Install ota-server, see https://github.com/aabadie/ota-server#installation
2. Start the otaserver on your laptop (just for testing):

      $ cd <ota-server-src-dir>
      $ python3 otaserver/main.py --debug
3. Generate the keys, etc
4. Publish a new version:

      $ SUIT_MODE=http SUIT_OTA_SERVER_URL=http://localhost:8080 make BOARD=samr21-xpro -C examples/suit_update/ suit/publish

5. Notify one device of an update (assuming `<ipv6 addr>` is reachable and can be reached from the http server):

      $ SUIT_MODE=http SUIT_OTA_SERVER_URL=http://localhost:8080 SUIT_CLIENT=[<ipv6 addr>] make BOARD=samr21-xpro -C examples/suit_update/ suit/notify

### CODE REVIEW

**Comment by aabadie** [comment]
cc @emmanuelsearch and @fjmolinas

File: sys/include/net/sock/util.h [hunk 1]
```diff
@@ -104,7 +104,7 @@ bool sock_udp_ep_equal(const sock_udp_ep_t *a, const sock_udp_ep_t *b);
                                              limit on the string iterator */
 #define SOCK_HOSTPORT_MAXLEN    (64U)   /**< maximum length of host:port part for
                                              sock_urlsplit() */
-#define SOCK_URLPATH_MAXLEN     (64U)   /**< maximum length path for
+#define SOCK_URLPATH_MAXLEN     (128U)  /**< maximum length path for
```
Comment (fjmolinas) [inline sys/include/net/sock/util.h]: I think now it is using blockwise to get the url, this shouldn't be needed.

File: examples/suit_update/Makefile [hunk 2]
```diff
@@ -61,12 +61,14 @@ QUIET ?= 1
 
 ifeq (,$(filter native,$(BOARD)))
   GNRC_NETIF_NUMOF := 2
-  USEMODULE += ethos
-  USEMODULE += gnrc_uhcpc
+  ifeq (local, $(SUIT_MODE))
```
Comment (fjmolinas) [inline examples/suit_update/Makefile]: I don't feel this is a correct representation of the choice base. I can perfectly use the http server and ethos.

**Comment by fjmolinas** [comment]
@aabadie @kaspar030 Tested using the default (local) mode and using http, works in both cases,

Only issue, in the above instructions it should be SUIT_CLIENT=[] and not OTA_CLIENT.

**Comment by aabadie** [comment]
Thanks for having a look @fjmolinas. I updated the application Makefile and dropped the sock util change, as you suggested.

**Pull request closed without merging**